File: algorithms/1979.find-greatest-common-divisor-of-array/solution.go
func findGCD(nums []int) int {
	min := 1001
	max := 0

	for _, n := range nums {
		if n < min {
			min = n
		}

		if n > max {
			max = n
		}
	}

	return gcd(max, min)
}

func gcd(x int, y int) int {
	if y == 0 {
		return x
	}

	return gcd(y, x%y)
}

File: algorithms/1979.find-greatest-common-divisor-of-array/solution.rs
impl Solution {
    pub fn find_gcd(nums: Vec<i32>) -> i32 {
        fn gcd(x: i32, y: i32) -> i32 {
            if y == 0 {
                return x;
            }

            gcd(y, x % y)
        }

        let mut min = 1001;
        let mut max = 0;
        for n in nums {
            if n < min {
                min = n;
            }

            if n > max {
                max = n;
            }
        }

        gcd(max, min)
    }
}
